### ISSUE TITLE
Second comment line is not indented correctly

### ISSUE THREAD

**Issue by davidBar-On** [rust-lang/rustfmt#4551]
**Input**

```rust
fn main() {
    let x = 111;   /* First Comment line 1
                    * First Comment line 2 */

    let x = 222;   /* Second Comment line 1
		    * Second Comment line 2 */
}
```

**Output**

```rust
fn main() {
    let x = 111; /* First Comment line 1
                  * First Comment line 2 */

    let x = 222; /* Second Comment line 1
                    * Second Comment line 2 */
}
```

**Expected output**

```rust
fn main() {
    let x = 111; /* First Comment line 1
                  * First Comment line 2 */

    let x = 222; /* Second Comment line 1
                  * Second Comment line 2 */
}
```

The issue is the **extra indentation of line 2 of the Second comment**.  The reason is that the `*` in that line is preceded by a `tab` and **not** by a `space`, and that [light_rewrite_comment()](https://github.com/rust-lang/rustfmt/blob/30bda450be1c8ded7b0eda63838b89cdbe3594db/src/formatting/comment.rs#L956-L957) is copying this preceding character from the original code.

If this is **regarded as a bug** that I can submit a PR with a fix from PR #4518 (simple fix - forcing the character preceding the `*` to be a `space`).

**Meta**

- rustfmt version: `rustfmt 2.0.0-rc.2-nightly (30bda450 2020-11-18)`
- From where did you install rustfmt?: self build

**Comment by calebcartwright** [comment]
I'm having difficulty reproducing this on master. Do you have any non-default config options set?

**Comment by davidBar-On** [comment]
Sorry, my mistake.  Somehow I wrongly formatted the input example in GitHub.  No special option is set. A right input with tab size 4 is:
```rust
fn main() {
   /* Comment1 line 1
    * Comment1 line 2 */
    let y1 = 6;/* Comment2 line 1
		* Comment2 line 2 */
}
````
In this example, both **input** comments start **one `space` before tab alignment** (e.g. 3 spaces before the first comment), and the second line in both has a `tab` before the `*`.  It seems that the issue happens when the `*` in the second line is preceded by a `tab` in the input and in the output the `*` is not aligned with tab spaces.

**Comment by calebcartwright** [comment]
Just to clarify, are you saying that in order to reproduce that the comment itself needs to _contain_ a hard tab?

**Comment by davidBar-On** [comment]
Yes, for the problem to happen a non-first line of a comment in the input should have a hard tab just before the `*`.  In addition, in the output the `*` should not be aligned with tab spaces, so when the hard tab is copied from the input it aligns the `*` incorrectly.

**Comment by ytmimi** [comment]
I haven't checked if this is still an issue with the `2.x branch`, but with `rustfmt 1.5.1-nightly (9c8ccc32 2022-07-26)` this is how the input gets formatted.

```rust
fn main() {
    let x = 111; /* First Comment line 1
     * First Comment line 2 */

    let x = 222; /* Second Comment line 1
     * Second Comment line 2 */
}
```

**Comment by davidBar-On** [comment]
Back-porting of PR #4658 (which I will submit soon) resolves the original issue of the second comment line indentation when it is preceded by a tab.

> I haven't checked if this is still an issue with the `2.x branch`, but with `rustfmt 1.5.1-nightly (9c8ccc32 2022-07-26)` this is how the input gets formatted.

 This is a `version Two` issue.  In version Two, the second line of a comment that starts on the "same line" is indented per the including line indentation and not per the comment start.  This works well for `\\` comments but it is not clear whether it is o.k. for the `/*`, as the above example shows.

**The question is which of the following formatting should be the output of version 2** -

Current version One formatting:
```rust
fn main() {
    let x = 111; /* First Comment line 1
                  * First Comment line 2
                  * First Comment line 3
                  */
    // Following first comment
}
```
Or current version Two formatting:
```rust
fn main() {
    let x = 111; /* First Comment line 1
     * First Comment line 2
     * First Comment line 3
     */
    // Following first comment
}
````

**Comment by ytmimi** [comment]
The version One formatting seems right to me. I think we should align block comments. Do multi-line, adjacent line comments also get aligned in version One?

**Comment by davidBar-On** [comment]
If I understood correctly what is meant by "multi-line, adjacent line comments", following is version One output (no tabs):
```rust
fn main() {
    let x = 222; /* Second Comment line 1
                  * Second Comment line 2
                  * Second Comment line 3
                  */
    /* Following second comment line 1
     * Following second comment line 2
     * Following second comment line 3
     */
}
```

**Comment by ytmimi** [comment]
Sorry, Yeah, that's a super confusing sentence now that I'm reading it back to myself. I mean do we indent line comments (`//`) that follow one another together?

```rust
fn main() {
    let x = 222; /* Second Comment line 1
                  * Second Comment line 2
                  * Second Comment line 3
                  */
    let y = 333; // Following second comment line 1
                 // Following second comment line 2
                 // Following second comment line 3
}
```

**Comment by davidBar-On** [comment]
Yes, this is the output of version one.

In version Two the output is:
```rust
fn main() {
    let x = 222; /* Second Comment line 1
     * Second Comment line 2
     * Second Comment line 3
     */
    let y = 333; // Following second comment line 1
    // Following second comment line 2
    // Following second comment line 3
}
```

My understanding is that the formatting of the consecutive one line indentation is the correct indentation (by design) for version Two, and **the question is whether the multi-line indentation is correct also, or should it be as for version One**.

(By the way, I assume the version Two indentation of the one line comments is the desired indentation, since in many cases the second (and on) comments will be to the next statemen.)

**Comment by davidBar-On** [comment]
Submitted PR #5480 with a suggested fix for this issue.  In version Two it indents the multiline comments according to the first line.
